sia/core.go

```diff
@@ -36,31 +55,70 @@ type Core struct {
 // createCore creates a server, host, miner, renter and wallet and
 // puts it all in a single environment struct that's used as the state for the
 // main package.
-//
-// TODO: swap out the way that CreateCore is called so that the wallet,
-// host, etc. can all be used as input - or not supplied at all.
-func CreateCore(hostDir string, walletFile string, serverAddr string, nobootstrap bool) (c *Core, err error) {
+func CreateCore(config Config) (c *Core, err error) {
+	if config.Host == nil {
+		err = errors.New("cannot have nil host")
+		return
+	}
+	if config.HostDB == nil {
+		err = errors.New("cannot have nil hostdb")
+		return
+	}
+	if config.Miner == nil {
+		err = errors.New("cannot have nil miner")
+		return
+	}
+	if config.Renter == nil {
+		err = errors.New("cannot have nil renter")
+		return
+	}
+	if config.Wallet == nil {
+		err = errors.New("cannot have nil wallet")
+		return
+	}
+
+	// Set the number of procs equal to the number of cpus.
+	runtime.GOMAXPROCS(runtime.NumCPU())
```

Comment:
I was really unsure where to put it. Main? Daemon? But it seemed to me like Core is the package that's calling the shots, so it should be manipulating the runtime environment to suit itself.